drivers/src/sha256.rs

```diff
@@ -137,21 +113,14 @@ impl Sha256 {
     /// operations will not be used after this function is called.
     ///
     /// This function is safe to call from a trap handler.
-    pub unsafe fn zeroize() {
-        let mut sha256 = Sha256Reg::new();
-        sha256.regs_mut().ctrl().write(|w| w.zeroize(true));
-    }
+    fn zeroize_internal(&mut self);
 
     /// Copy digest to buffer
     ///
     /// # Arguments
     ///
     /// * `buf` - Digest buffer
-    fn copy_digest_to_buf(&mut self, buf: &mut Array4x8) -> CaliptraResult<()> {
-        let sha256 = self.sha256.regs();
-        *buf = Array4x8::read_from_reg(sha256.digest());
-        Ok(())
-    }
+    fn copy_digest_to_buf(&mut self, buf: &mut Array4x8) -> CaliptraResult<()>;
```

Comment:
I don't like how this formerly private function is now exposed to the world.

As an experiemnt, I tried making this refactor as minimal as possible:
https://github.com/chipsalliance/caliptra-sw/pull/731

The main downside to my approach is that the fuzz tests don't cover any of the additional logic in the sha256 driver.

Reply:
The experiment has been merge into this PR.